### ISSUE TITLE
Tutorial database

### ISSUE THREAD

**Issue by rdstern** [opencdms/Climsoft#268]
Samuel confirmed that CLIMSOFT, when installed, is automatically provided with an empty database plus a tutorial-style database that can be used for training purposes.

If useful we also have data from 54 rainfall stations and 6 stations that measure other variables.  If needed they providers have given permission that these data can be shared.  Hence they could be made available within the tutorial database, or as a second training database if needed.  The data used to be in CLIMSOFT 3.2, which would have been ideal to practice the transfer.  Unfortunately the machine with the database is no more.  So we have the data fgiles and the station information, but not the backed-up database.

**Comment by rdstern** [comment]
[WesternKenya.zip](https://github.com/opencdms/Climsoft/files/1549353/WesternKenya.zip)

Please could we swap the possible tutorial files for CLIMSOFT?  Here are the files for Western Kenya.  There are 3 files, one with the station information, the second with the rainfall data (600,000 records) for the 52 stations and the third with the information from the 6 stations that had multiple elements.  In R-Instat these 3 files can be saved together and that file is therefore also included.

I would also like to be able to use the standard CLIMSOFT tutorial database in R -Instat.

**Comment by mhabimana** [comment]
@rdstern - Thanks for sharing the data. Can you please describe the column of the file  "All_data1.scv" by their element name?(e.g.what stands for column x2,x3,x4,........).

**Comment by rdstern** [comment]
@mhabimana I would be happy to do that when I have the response to my earlier query on the meaning of the different elements.  Meanwhile X2 is element 2 in CLIMSOFT i.e. Tmax etc.  That was the only safe way I could see of doing this.

**Comment by mhabimana** [comment]
@rdstern - Kindly check my reply at #270 (I attached the observation element list including their descriptions)

**Comment by rdstern** [comment]
OK, now I see you have provided the useful list of elements.  I am still a bit unsure of the difference between element 17 relative humidity and 105, in stations that measure just once a day.  Is it a different instrument that is used?

Wind speed 56 and 111 is simply the units, i.e. metres/sec and knots.  I see that now.  But the mean of the data in element 56 is just 2.8 metres per second, while the minimum of the data in element 111 is 0, mean is 108 and maximum is 493.  That's odd, because:
a) In element 111 you say the minimum is 1 and the maximum is 30?
b) The conversion from m/sec to knots is about 1 to 2.  So I still don't understand those 2 elements.  That's why I persisted with the element numbers in the file.  I don't want to put elements together until I am sure they are measuring the same thing.

**Comment by mhabimana** [comment]
@rdstern - It makes sense to combine the attached data with the existing sample data so that wide range of data is available for practice.

**Comment by smachua** [comment]
@rdstern - In simple terms, Climsoft metadata defines element code 17 as the daily RH for values likely* computed from hourly observations that are described by element code 105. Similarly element 56 is the daily wind speed average likely* from hourly wind speed described by element code 111.
I have not seen the data you're referring to neither its source, so I may not comment on the values described therein.
* Daily average values may also be computed from AWS data whose period of observation is in minutes.

**Comment by mhabimana** [comment]
@rdstern -Most of the case 111 is estimated in knots using beaufaurt scale and the lower limit in climsoft is 0knots and the upper limit is 30 knots(for QC purpose i.e values above 30 knots need much attention. In your case ,I am not sure if the values you have under element 56 were converted from 111 or were measured or calculated by another means.

**Comment by mhabimana** [comment]
@smachua - The metadata information  shared by @rdstern has got only the start year and end year (there is no Day and Month). The import metadata open and closing date in V4 is using a calendar and will need the Day and Month to be able to import the metadata.This issue was raised by Cathy some times back,we need to see how to address such format in climsoft V4 metadata.

**Comment by Fortunata** [comment]
We need to be carefull when defining element codes which require scrutiny especially with the staff of the any particular NMHS . Through training or clear explanation is required in the tutorials to avoid confusion on the  elements being captured by the met service. For example in the elements list all elements 1-100 were attributed to daily time lines; then from 100 to 200 were attributed to hourly and 200 were for monthly.  i.e 5 is daily RF , 104 is hourly RF and 208 is monthly this was inherited from CLICOM .Samuel remember the challenge we had in Uganda over elements definition.

**Comment by mhabimana** [comment]
@Fortunata - Thanks for for your good recommendation. Could you please share with us the documentation, if available, on how to categorise the elements codes and stations codes inherited from CLICOM as you mentioned? This document can be very helpful as a reference to us and to future generations.

**Comment by Fortunata** [comment]
Still on this issue Rogers element 17 for RH is a daily value mean if at all the daily values from the available hours are averaged that is elem 17 but cation must be taken if only few hrs are available however its appropriate to average the two hours of 0600Z and 1200Z most available in the NMHS,
The elem 105 simply refers to the hour at which the RH was recorded and some users may want to know the dianal behavior of RH when it is high in the morning and low in the afternoon

**Comment by rdstern** [comment]
At the start of this discussion we noted "Samuel confirmed that CLIMSOFT, when installed, is automatically provided with an empty database **plus a tutorial-style database that can be used for training purposes**."

With the new CLIMSOFT I would like to use the tutorial-style database to test the current facilities.  Maxwell says he doesn't have it yet.  Could you send it or add it here, so it is available to us all.

Thanks

**Comment by maxwellfundi** [comment]
@isedwards  yeah I have both of these. Will try run the test one from an older file that had the data which i think will be useful for @rdstern . The test database we have is empty because we cleared the data while we were testing the data entry forms.

**Comment by maxwellfundi** [comment]
@mhabimana @isedwards I have ran the older sql file, and I still got the two databases mariadb_climsoft_db_v4 and mariadb_climsoft_test_db_v4 and none seems to have the data into it. the test one has a few rows in the monthly table but nothing more. @smachua Could you possibly send us a copy of the test database that contains some substantial set of test data?

**Comment by mhabimana** [comment]
@maxwellfundi: As you can see below,the observationfinal database table for the mariadb_test_db has got data:
![image](https://user-images.githubusercontent.com/10340170/41916803-7c7d4880-7961-11e8-9328-7783b8dd9b8c.png)

**Comment by maxwellfundi** [comment]
@mhabimana  thanks alot, I got it!

**Comment by smachua** [comment]
@mhabimana @isedwards I attached script for the test database
[mariadb_climsoft_test_db_v4.txt](https://github.com/opencdms/Climsoft/files/2138352/mariadb_climsoft_test_db_v4.txt)
